core/templates/pages/exploration-editor-page/translation-tab/translation-tab.component.spec.ts

```diff
@@ -402,6 +402,10 @@ describe('Translation tab component', function() {
 
   it('should finish translation tutorial when welcome translation modal is' +
     ' dismissed', fakeAsync(() => {
+    spyOn(userExplorationPermissionsService, 'getPermissionsAsync').and
```

Comment:
I've checked all the instances of Context Service to check whether they are properly mocked/spied correctly or not surprisingly all the instances are mocked correctly.
Then I've looked into the call stacks of Translation Tab Component,
in `$onInit` we are using `UserExplorationPermissionsService.getPermissionsAsync` which is not mocked in this test. (It has been mocked in other tests in the same suite).
Call Stack: `$onInit` --> `UserExplorationPermissionsService.getPermissionsAsync` --> `ExplorationPermissionsBackendApiService.getPermissionsAsync` --> `ContextService.getExplorationId`
Since spec file is in hybrid state, even though `ContextService.getExplorationId` has been mocked it is calling the original service.
By spyin on `userExplorationPermissionsService.getPermissionsAsync` we are terminating the call stack Hence, this will fix the issue

Reply:
Looks like `UserExplorationPermissionsService.getPermissionsAsync` call gets triggered when `$scope.$apply();` gets called. Can you add the above comment to the PR description?

Reply:
> Since the spec file is in a hybrid state, even though ContextService.getExplorationId has been mocked it is calling the original service.

Is that the reason why it fails sporadically and not every time?